d3a_interface/matching_algorithms/pay_as_clear_matching_algorithm.py

```diff
@@ -119,48 +118,46 @@ def _clearing_point_from_supply_demand_curve(bids_rate_energy: Dict, offers_rate
             if len(clearing) > 0:
                 return clearing[-1].rate, clearing[-1].energy
 
-    def get_clearing_point(self, bids: List[Dict], offers: List[Dict], current_time):
+    def get_clearing_point(self, bids: List[Dict], offers: List[Dict], current_time, market_id):
         self.sorted_bids = sort_list_of_dicts_by_attribute(bids, "energy_rate", True)
         self.sorted_offers = sort_list_of_dicts_by_attribute(offers, "energy_rate")
-        clearing = None
+        clearing, cumulative_bids, cumulative_offers = None, None, None
 
         if len(self.sorted_bids) == 0 or len(self.sorted_offers) == 0:
             return
 
         if ConstSettings.IAASettings.PAY_AS_CLEAR_AGGREGATION_ALGORITHM == 1:
             cumulative_bids = self._accumulated_energy_per_rate(self.sorted_bids)
             cumulative_offers = self._accumulated_energy_per_rate(self.sorted_offers)
-            self.state.cumulative_bids = cumulative_bids
-            self.state.cumulative_offers = cumulative_offers
             ascending_rate_bids = OrderedDict(reversed(list(cumulative_bids.items())))
             clearing = self._clearing_point_from_supply_demand_curve(
                 ascending_rate_bids, cumulative_offers)
         elif ConstSettings.IAASettings.PAY_AS_CLEAR_AGGREGATION_ALGORITHM == 2:
             cumulative_bids = self._discrete_point_curve(self.sorted_bids, math.floor)
             cumulative_offers = self._discrete_point_curve(self.sorted_offers, math.ceil)
-            max_rate = self._populate_market_cumulative_offer_and_bid(cumulative_bids,
-                                                                      cumulative_offers)
-            clearing = self._get_clearing_point(max_rate)
+            max_rate, cumulative_bids, cumulative_offers = \
+                self._populate_market_cumulative_offer_and_bid(cumulative_bids, cumulative_offers)
+            clearing = self._get_clearing_point(max_rate, cumulative_bids, cumulative_offers)
```

Comment:
`_get_clearing_point` returns two values, but I dont see it is being used here. I think you can remove it (`cumulative_bids[rate]`) then. Right?

Reply:
Its saved to `clearing ` as tuple i.e first as clearing rate and second as clearing energy. Its been like that before this PR too.